src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Strings.vb

```diff
@@ -3,48 +3,280 @@
 Imports Microsoft.VisualBasic.CompilerServices.Utils
 
 Namespace Global.Microsoft.VisualBasic
-    <Global.System.Diagnostics.DebuggerNonUserCode()>
+
     Public Module Strings
-        Public Function ChrW(charCode As Integer) As Char
-            If charCode < -32768 OrElse charCode > 65535 Then
-                Throw New Global.System.ArgumentException()
+
+        '============================================================================
+        ' Character manipulation functions.
+        '============================================================================
+        Public Function Asc(ByVal [String] As Char) As Integer
+
+            'The IConvertible.ToInt32 implementation on Char
+            '   just calls Convert.ToInt32()
+            Dim CharValue As Integer = Convert.ToInt32([String])
+
+            If CharValue < 128 Then
+                Return CharValue
             End If
-            Return Global.System.Convert.ToChar(charCode And &HFFFFI)
+
+            Try
+                Dim enc As Encoding
+                Dim b() As Byte
+                Dim c() As Char
+                Dim iByteCount As Integer
+
+                enc = GetFileIOEncoding()
+
+                c = New Char() {[String]}
+
+                If enc.IsSingleByte Then
+                    'SBCS
+                    b = New Byte(0) {}
+                    iByteCount = enc.GetBytes(c, 0, 1, b, 0)
+                    Return b(0)
+                End If
+
+                'DBCS char
+                b = New Byte(1) {}
+                iByteCount = enc.GetBytes(c, 0, 1, b, 0)
+                If iByteCount = 1 Then
+                    Return b(0)
+                End If
+                If BitConverter.IsLittleEndian Then
+                    'Swap the bytes since storage is big-endian
+                    Dim byt As Byte
+                    byt = b(0)
+                    b(0) = b(1)
+                    b(1) = byt
+                End If
+                Return BitConverter.ToInt16(b, 0)
+
+            Catch ex As Exception
+                Throw ex
+
+            End Try
+
         End Function
-        Public Function AscW([string] As String) As Integer
-            If ([string] Is Nothing) OrElse ([string].Length = 0) Then
-                Throw New Global.System.ArgumentException()
+
+        Public Function Asc(ByVal [String] As String) As Integer
+
+            If ([String] Is Nothing) OrElse ([String].Length = 0) Then
+                Throw New ArgumentException(SR.Argument_LengthGTZero1, NameOf([String]))
             End If
-            Return AscW([string].Chars(0))
+
+            Dim ch As Char = [String].Chars(0)
+            Return Asc(ch)
+
         End Function
-        Public Function AscW([string] As Char) As Integer
-            Return AscW([string])
+
+        Public Function AscW([String] As String) As Integer
+            If ([String] Is Nothing) OrElse ([String].Length = 0) Then
```

Comment:
string.IsNullOrEmpty?

Reply:
I'm expecting this code is pretty much a dead end after it's ported so it seemed best to leave it functionally unchanged lest I introduce some new bug (particularly since to a C# developer there are some gotchas in VB)